src/PowerShellEditorServices.Engine/Services/TextDocument/CompletionResults.cs

```diff
@@ -0,0 +1,341 @@
+        /// <returns>True if the CompletionResults instances have the same details.</returns>
+        public override bool Equals(object obj)
+        {
+            CompletionDetails otherDetails = obj as CompletionDetails;
+            if (otherDetails == null)
+            {
+                return false;
+            }
+
+            return
+                string.Equals(this.CompletionText, otherDetails.CompletionText) &&
+                this.CompletionType == otherDetails.CompletionType &&
+                string.Equals(this.ToolTipText, otherDetails.ToolTipText) &&
+                string.Equals(this.SymbolTypeName, otherDetails.SymbolTypeName);
+        }
+
+        /// <summary>
+        /// Returns the hash code for this CompletionResults instance.
+        /// </summary>
+        /// <returns>The hash code for this CompletionResults instance.</returns>
+        public override int GetHashCode()
+        {
+            return
+                string.Format(
+                    "{0}{1}{2}{3}{4}",
+                    this.CompletionText,
+                    this.CompletionType,
+                    this.ListItemText,
+                    this.ToolTipText,
+                    this.SymbolTypeName).GetHashCode();
+        }
+
+        #endregion
+
+        #region Private Methods
+
+        private static CompletionType ConvertCompletionResultType(
+            CompletionResultType completionResultType)
+        {
+            switch (completionResultType)
+            {
+                case CompletionResultType.Command:
+                    return CompletionType.Command;
+
+                case CompletionResultType.Method:
+                    return CompletionType.Method;
+
+                case CompletionResultType.ParameterName:
+                    return CompletionType.ParameterName;
+
+                case CompletionResultType.ParameterValue:
+                    return CompletionType.ParameterValue;
+
+                case CompletionResultType.Property:
+                    return CompletionType.Property;
+
+                case CompletionResultType.Variable:
+                    return CompletionType.Variable;
+
+                case CompletionResultType.Namespace:
+                    return CompletionType.Namespace;
+
+                case CompletionResultType.Type:
+                    return CompletionType.Type;
+
+                case CompletionResultType.Keyword:
+                case CompletionResultType.DynamicKeyword:
+                    return CompletionType.Keyword;
+
+                case CompletionResultType.ProviderContainer:
+                    return CompletionType.Folder;
+
+                case CompletionResultType.ProviderItem:
+                    return CompletionType.File;
+
+                default:
+                    // TODO: Trace the unsupported CompletionResultType
+                    return CompletionType.Unknown;
```

Comment:
How do we feel about making `LangVersion` `preview`?

```csharp
return completionResultType switch
{
    CompletionResultType.Command => CompletionType.Command,
    CompletionResultType.Method => CompletionType.Method,
    CompletionResultType.ParameterName => CompletionType.ParameterName,
    CompletionResultType.ParameterValue => CompletionType.ParameterValue,
    CompletionResultType.Property => CompletionType.Property,
    CompletionResultType.Variable => CompletionType.Variable,
    CompletionResultType.Namespace => CompletionType.Namespace,
    CompletionResultType.Type => CompletionType.Type,
    CompletionResultType.Keyword => CompletionType.Keyword,
    CompletionResultType.DynamicKeyword => CompletionType.Keyword,
    CompletionResultType.ProviderContainer => CompletionType.Folder,
    CompletionResultType.ProviderItem => CompletionType.File,
    // TODO: Trace the unsupported CompletionResultType
    _ => CompletionType.Unknown,
};
```

Alternatively since it's mostly one to one you could have a map array, e.g.

```csharp
private static readonly s_completionTypeMap =
{
    CompletionType.Command,
    // etc
};

void Access(CompletionResultType resultType) => s_completionTypeMap[(int)resultType];
```